OutOfSchool/OutOfSchool.WebApi/Services/PermissionsForRoleService.cs

```diff
@@ -0,0 +1,108 @@
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Extensions;
+using OutOfSchool.WebApi.Models;
+
+namespace OutOfSchool.WebApi.Services
+{
+    public class PermissionsForRoleService : IPermissionsForRoleService
+    {
+        private readonly IEntityRepository<PermissionsForRole> repository;
+        private readonly ILogger<PermissionsForRoleService> logger;
+        private readonly IStringLocalizer<SharedResource> localizer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PermissionsForRoleService"/> class.
+        /// </summary>
+        /// <param name="repository">Repository.</param>
+        /// <param name="logger">Logger.</param>
+        /// <param name="localizer">Localizer.</param>
+        public PermissionsForRoleService(IEntityRepository<PermissionsForRole> repository, ILogger<PermissionsForRoleService> logger, IStringLocalizer<SharedResource> localizer)
+        {
+            this.localizer = localizer;
+            this.repository = repository;
+            this.logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public async Task<IEnumerable<PermissionsForRoleDTO>> GetAll()
+        {
+            logger.LogInformation("Getting all Permissions for Roles started.");
+
+            var permissionsForRoles = await repository.GetAll().ConfigureAwait(false);
+
+            logger.LogInformation(!permissionsForRoles.Any()
+                ? "PermissionsForRole table is empty."
+                : $"All {permissionsForRoles.Count()} records were successfully received from the PermissionsForRole table");
+
+            return permissionsForRoles.Select(permissionsForRoles => permissionsForRoles.ToModel()).ToList();
+        }
+
+        /// <inheritdoc/>
+        public async Task<PermissionsForRoleDTO> GetByRole(string roleName)
+        {
+            logger.LogInformation($"Getting Permissions for role by roleName started. Looking role = {roleName}.");
+
+            var permissionsForRole = await repository.GetByFilter(p => p.RoleName == roleName).ConfigureAwait(false);
+            if (permissionsForRole == null)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(roleName),
+                    localizer["There are no packed permissions for role with such name"]);
+            }
+
+            logger.LogInformation($"Successfully got a permissionsForRole with name  {permissionsForRole.FirstOrDefault().RoleName}.");
+
+            return permissionsForRole.FirstOrDefault().ToModel();
+        }
+
+        /// <inheritdoc/>
+        public async Task<PermissionsForRoleDTO> Create(PermissionsForRoleDTO dto)
+        {
+            logger.LogInformation($"Permissions for Role - {dto?.RoleName} creating was started.");
+
+            var allPermissions = await repository.GetAll().ConfigureAwait(false);
```

Comment:
Should we check dto for null not to make a request in this case? You'll get an exception after it. And do we need to return allPermisions into our memory, you can check it in a DB.